Changes to Application_Templates routing

enragednuke opened the pull request:
* Now is `/applications/:id`
 * Can also find templates by using the format `/application/:department/:position`

Closes #190 

(send help I don't know how to work codeclimate)

(yes I know I pushed to my fork's master on accident)

dfaulken commented:
Codeclimate is broken; see #186. If Travis passes you're good after someone looks at your code.

dfaulken commented on app/controllers/application_templates_controller.rb:
```diff
@@ -27,7 +27,24 @@ def toggle_active
 
   def find_template
-    params.require :id
+    if params[:specific_path]
```
Right, so this should be done via routing, not with controller method changes.

http://guides.rubyonrails.org/routing.html#dynamic-segments

enragednuke replied:
I'm confused, isn't that exactly what I'm doing? But you need that additional parameter to be able to distinguish between WHICH route you're taking so you can specify which parameters to require. You can't just require all of them, because then both will fail to missing the other.

dfaulken replied:
Yes, my bad. I misread your code. You don't need to preserve the original routing, which will simplify a lot of the code. You can assume it follows the department/position format - hence my comment in the issue about being careful to update the hardlink on the public site when we roll this out.

enragednuke replied:
Gotcha. I'll swap this stuff out when I get in tomorrow.

enragednuke commented:
@dfaulken A bunch of tests exploded once I made it dept/pos only routing, but they're all fixed and this should be good to go unless there's an issue with any of the code as of now.

enragednuke commented:
Should've addressed all the issues mentioned with the latest push.

dfaulken commented on app/helpers/application_templates_helper.rb:
```diff
@@ -0,0 +1,2 @@
+module ApplicationTemplatesHelper
```
Can delete this file

dfaulken commented:
Looks good - I'll take care of the file trim. Thanks for your work on this @enragednuke

enragednuke commented:
@dfaulken So I found a side-effect of this. This ended up affecting all the routes to toggle_active as well (since it uses `find_template` as well) so I need to edit all of the `toggle_active_application_template_url` links too and push that.

dfaulken commented:
Interesting - why wasn't this caught by tests?

enragednuke commented:
I don't think the tests were testing clicking on those links. I know I modified some tests in regards to it because they were failing when I modified RSpec.
